### PR TITLE
Xattr test fixes

### LINKED ISSUE #254: The tests in xattr.c assume an FS with xattr support
The tests in `tests/preload/xattr/xattr.c` expect `setxattr` to work on paths not mounted under pmemfile.
This caused a problem when I had `TEST_DIR=/dev/shm`:
```
$ cf -R preload_xa
Test project /home/tej/code/pmemfile/build
    Start 68: preload_xattr
1/1 Test #68: preload_xattr ....................***Failed    0.09 sec
preload_xattr: setxattr failed: Operation not supported
CMake Error at /home/tej/code/pmemfile/tests/helpers.cmake:105 (message):
  /home/tej/code/pmemfile/build/tests/preload/preload_xattr /dev/shm//xattr
  failed: 3
Call Stack (most recent call first):
  /home/tej/code/pmemfile/tests/preload/xattr/xattr.cmake:45 (execute)
```
This test failure looks exactly as if libpmemfile would return ENOTSUP, but in this case the kernel returned this error code.

### PR DESCRIPTION
**Pull request by GBuella**
Closes #254

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/257)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -259,6 +259,7 @@ endif()
 
 if(DEVELOPER_MODE)
 	add_flag(-Werror)
+	add_flag(-Wno-unused-command-line-argument)
```
Comment (marcinslusarz) [inline CMakeLists.txt]: What is this for?

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 2 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/257#-:-KpFq1OOPMqs3ZE0kY2l:bnfp4nl)*

**Comment by GBuella** [comment]
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[CMakeLists.txt, line 262 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/257#-KpFmnf4PVDaoHWvk_KG-r1-262:-KpJxzoVEcWQAdzVi6gv:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/42b4f8115f707fda2fcf95813a9b2fc0f4f90c66/CMakeLists.txt#L262)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

What is this for?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/257)*

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/257#-:-KpJy6z-Z68uRYfi2576:bnfp4nl)*

**Comment by krzycz** [comment]
:lgtm:

---

Reviewed 1 of 3 files at r1, 1 of 2 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/257#-:-KpKMdL9QQOdgXejIOi4:bnfp4nl)*

**Pull request merged**